docs/proposals/20240807-in-place-updates-implementation-notes.md

```diff
@@ -86,3 +86,173 @@ sequenceDiagram
     MS2 (NewMS)-->>MS Controller: Yes, M1!
     MS Controller->>M1: Remove annotation ".../pending-acknowledge-move": ""
 ```
+
+## Notes about in-place update implementation for KubeadmControlPlane
+
+- In-place updates respect the existing control plane update strategy:
+  - KCP controller uses `rollingUpdate` strategy with `maxSurge` (0 or 1)
+  - When `maxSurge` is 0, no new machines are created during rollout - only in-place updates or scale down
+  - When `maxSurge` is 1:
+    - Controller first scales up by creating one new machine to maximize fault tolerance
+    - Once at `maxReplicas` (desiredReplicas + 1), evaluates whether to in-place update or scale down old machines
+    - For each old machine needing rollout: if eligible for in-place update, performs in-place; otherwise scales down
```

Comment:
Best to break this down into multiple sentences

e.g. for each machine it evaluates if it is eligible for in-place update. If so, it.... , otherwise ....